backend/src/main/java/com/dccsacademy/repositories/SupplierRepository.java

```diff
@@ -0,0 +1,31 @@
+package com.dccsacademy.repositories;
+
+import com.dccsacademy.entities.SupplierEntity;
+import io.quarkus.hibernate.orm.panache.PanacheRepository;
+import jakarta.enterprise.context.ApplicationScoped;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@ApplicationScoped
+public class SupplierRepository implements PanacheRepository<SupplierEntity> {
+
+    public List<SupplierEntity> searchSuppliers(String id, String name, String city) {
+        name = (name != null) ? name.trim().toLowerCase() : "";
```

Comment:
1. Make util method that prepares your strings for search (% + lower case value + %). Call that instead of duplicating code. 
2. Write query using Criteria API instead of using hardcoded names (name, city).
3. id is Long, so accept it as Long not as String.